src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java

```diff
@@ -117,6 +111,24 @@ private void buildSscsDocumentFromScan(SscsCaseData sscsCaseData, State caseStat
 
     }
 
+    private void checkWarningsAndErrors(ScannedDocument scannedDocument, String caseId, boolean ignoreWarnings) {
+        if (scannedDocument.getValue().getUrl() == null) {
+            preSubmitCallbackResponse.addError("No document URL so could not process");
+        }
+
+        if (scannedDocument.getValue().getFileName() == null) {
+            preSubmitCallbackResponse.addError("No document file name so could not process");
+        }
+
+        if (equalsIgnoreCase(scannedDocument.getValue().getType(), COVERSHEET)) {
+            if (!ignoreWarnings) {
+                preSubmitCallbackResponse.addWarning("Coversheet will be ignored, are you happy to proceed?");
```

Comment:
Message doesn't look right, as if ignoreWarning is false then message says conversheet will be ignored

Reply:
Yeh that is correct. The way CCD works is the first time you press submit, the 'ignoreWarnings' flag in the payload is false. If we return a warning in our response to the user, they have the option of saying yes/no to ignore warning. Then, we receive the second payload, which will have the 'ignoreWarnings' as true, if they selected yes. In this case, we don't want to return another warning as they have already confirmed they are happy to proceed.